### PR TITLE
Airflow 2.8.0 has been released

### PR DESCRIPTION
**Pull request by ephraimbuddy**

### CODE REVIEW

File: README.md
Comment (Taragolis) [inline README.md:1]: Small nit:

PostgreSQL versions should be set `12, 13, 14, 15, 16` for either dev and stable
MySQL versions should be set `8.0, Innovation` for stable

**Review by potiuk — APPROVED** [review]
LGTM with applying the comments from @Taragolis

**Review by Taragolis — APPROVED** [review]

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,7 +21,204 @@
 
 .. towncrier release notes start
 
-Airflow 2.7.3 (2023-11-04)
+Airflow 2.8.0 (2023-12-14)
```
Comment (josh-fell) [inline RELEASE_NOTES.rst:24]: Should this be December 18th instead?

Comment (ephraimbuddy) [reply RELEASE_NOTES.rst:24]: I'll update it in a separate PR so we can cherrypick it to 2.8 branch and build docs

**Pull request merged**